components/Pagination/Pagination.js

```diff
@@ -19,26 +19,29 @@ const messages = defineMessages({
 });
 
 class Pagination extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.state = { pageValue: 0 };
     this.handleBlur = this.handleBlur.bind(this);
     this.handleChange = this.handleChange.bind(this);
   }
 
-  componentWillMount() {
+  componentDidMount() {
     this.setState({ pageValue: this.props.currentPage });
   }
 
-  componentWillReceiveProps(newProps) {
-    this.setState({ pageValue: newProps.currentPage });
-    // If the input has focus when the page value is updated, then select the
-    // text
+  componentDidUpdate() {
     if (this.paginationPage === document.activeElement) {
       this.paginationPage.select();
     }
   }
 
+  static getDerivedStateFromProps(newProps, prevState) {
+    if (prevState.pageValue !== newProps.currentPage) {
+      return { pageValue: newProps.currentPage };
+    } else return null;
```

Comment:
Does this work correctly? 
I would expect something like:
  return {}
for zero updates.

Reply:
I believe returning {} would not cause any updates to the state but per the docs:

> It should return an object to update the state, or null to update nothing

Reply:
Ok thanks